app/src/main/java/com/techcourse/DispatcherServletInitializer.java

```diff
@@ -4,9 +4,9 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import web.org.springframework.web.WebApplicationInitializer;
+import webmvc.org.springframework.web.servlet.mvc.DispatcherServlet;
```

Comment:
`DispatcherServletInitializer`에 대한 이야기는 이 코멘트로 남기겠습니다.
(이니셜라이저라고 칭하겠습니다.)

우선 저도 이니셜라이저를 mvc 모듈 안으로 옮기려고 시도하다가 말았는데요.
현재는 이니셜라이저가 app 모듈에 존재하는 것이 적절하다고 생각하고 있습니다.
그 이유는 이니셜라이저의 역할이 `DispatcherServlet`을 내 애플리케이션에 맞게 사용하도록 설정하는 것이라고 생각하기 때문입니다.

`DispatcherServlet`은 현재 다양한 핸들러를 처리할 수 있도록 인터페이스를 제공하고 있습니다. 우리는 이니셜라이저를 통해서 애플리케이션 내의 핸들러를 `DispatcherServlet`의 인터페이스에 맞게 등록하고 있습니다.

즉, 이니셜라이저는 스프링을 사용하며 우리의 애플리케이션에 맞게 `DispatcherServlet`을 사용하기 위해 필요한 것이므로 app 모듈 내에 존재하는 것이 적절해 보입니다.

하지만 우리가 지금까지 스프링으로 개발을 하며 애플리케이션 내에 이니셜라이저를 구현한 경험이 없는 것은, 스프링 부트가 이를 자동으로 해주기 때문이라고 하네요..

정확한 레퍼런스와 함께 의견을 제시하면 좋았겠지만,, 저도 아직 제대로 찾아보지 않아서 틀린 정보가 있을 수도 있다는 점.. 양해 부탁드립니다..
(그런가보다~ 하고 넘어가 주셔도 좋습니다)

Reply:
[스프링의 DispatcherServlet 공식 문서](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-servlet.html)를 보다가 다음과 같은 내용을 찾았습니다.

> Spring Boot follows a different initialization sequence. Rather than hooking into the lifecycle of the Servlet container, Spring Boot uses Spring configuration to bootstrap itself and the embedded Servlet container. Filter and Servlet declarations are detected in Spring configuration and registered with the Servlet container. For more details, see the [Spring Boot documentation](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-embedded-container).

번역을 해보니.. 스프링 부트에서는 내장된 서블릿 컨테이너를 사용한다고 합니다. 아마도 이러한 내장 설정 덕분에 우리는 서블릿을 설정한 적이 없는 것으로 보입니다!